sensor/common/sensor/component_queue.go

```diff
@@ -0,0 +1,68 @@
+package sensor
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/queue"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/metrics"
+)
+
+const (
+	// componentProcessTimeout is the maximum time allowed for a sensor component to process a message from Central
+	componentProcessTimeout = 30 * time.Second
+)
+
+type ComponentQueue struct {
+	component common.SensorComponent
+	q         *queue.Queue[*central.MsgToSensor]
+}
+
+func NewComponentQueue(component common.SensorComponent) *ComponentQueue {
+	componentName := prometheus.Labels{
+		metrics.ComponentName: component.Name(),
+	}
+	componentQueueOperations, err := metrics.ComponentQueueOperations.CurryWith(componentName)
+	utils.CrashOnError(err)
+	c := &ComponentQueue{
+		component: component,
+		q: queue.NewQueue(
+			queue.WithQueueName[*central.MsgToSensor](component.Name()),
+			queue.WithMaxSize[*central.MsgToSensor](env.RequestsChannelBufferSize.IntegerSetting()),
+			queue.WithCounterVec[*central.MsgToSensor](componentQueueOperations),
+			queue.WithDroppedMetric[*central.MsgToSensor](componentQueueOperations.With(prometheus.Labels{
+				metrics.Operation: "Drop",
```

Comment:
[nit] We have defined constants for metric operations [here](https://github.com/stackrox/stackrox/blob/6a15c1e89a503cddf0681d4f906e2f9b854b85b6/pkg/metrics/operations.go#L1) if you want to use them.